community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/util.py

```diff
@@ -1918,6 +1918,11 @@ def nodeset_map(self, hostnames: list):
         for node in hostnames:
             nodeset_map[self.node_nodeset_name(node)].append(node)
         return nodeset_map
+    
+    #TODO check to see if stdout/rstrip is necessary
+    @lru_cache(maxsize=1)
+    def job_information(self, job_id: int) -> str:
```

Comment:
nit. `s/job_information/job/`

Let's do all the parsing in this function and return nice `dataclass`.

Reply:
Added, if this class structure isn't what you had in mind please re-open this convo.